ffmpeg/lpms_ffmpeg.c

```diff
@@ -1335,6 +1357,7 @@ int transcode(struct transcode_thread *h,
     has_frame = lpms_ERR_PACKET_ONLY != ret;
 
     if (AVMEDIA_TYPE_VIDEO == ist->codecpar->codec_type) {
+      if (ictx->flushing) dframe->pkt_dts = ictx->first_pkt->dts;
```

Comment:
Curious - what does this fix? Maybe good to note this in a code comment.

Reply:
This is used to keep the [segment-ordering check](https://github.com/livepeer/lpms/blob/6e167971ec68573d2cdf47cbb0b163fbb3a981a7/ffmpeg/lpms_ffmpeg.c#L995) work same as it did before.

FFmpeg sets pkt_dts to the dts of the [packet triggering the frame output](https://ffmpeg.org/doxygen/trunk/structAVFrame.html#aa52951f35ec9e303d3dfeb4b3e44248a) and *not* the packet that was actually decoded to get the frame.

If we don't set it to first_pkt->dts when using NULL-flushing, the last_frame->pkt_dts is set to 0 for the flushed frames and the segment-ordering check fails.

I tried modifying the segment-ordering check but that caused other tests to fail, and fixing those triggered more. So for this PR I'll avoid going into that rabbit hole.

I'll add a comment for this as well.